src/common/template.test.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { expect } from "chai";
+import { Template } from "./template";
+import { fileSync } from "tmp";
+import { writeFileSync, readFileSync } from "fs-extra";
+import * as Handlebars from "handlebars";
+
+describe("Create template instance", () => {
+  const errMsg = "Error initializing template";
+  it("throws error when template file path is undefined", async () => {
+    return expect(() => new Template(undefined)).to.throw(errMsg);
+  });
+  it("throws error when template file path is empty", async () => {
+    return expect(() => new Template("")).to.throw(errMsg);
+  });
+  it("throws error when template file path is null", async () => {
+    return expect(() => new Template(null)).to.throw(errMsg);
+  });
+  it("throws error when the template file do not exist", async () => {
+    return expect(() => new Template("/tmp/template.hbs")).to.throw(errMsg);
+  });
+  it("throws error when the template file has no content", async () => {
+    const templateFile = fileSync({ postfix: ".hbs" });
+    return expect(() => new Template(templateFile.name)).to.throw(
+      "Invalid template content"
+    );
+  });
+  it("creates instance of template with a valid template file", async () => {
+    const templateFile = fileSync({ postfix: ".hbs" });
+    const templateContent = "Test";
+    writeFileSync(templateFile.name, templateContent);
+    const template = new Template(templateFile.name);
+    //verify template and its content
+    expect(template).to.be.an.instanceof(Template);
+    expect(template.content).to.equal(templateContent);
```

Comment:
```suggestion
    expect(template.content).to.equal("Test");
```
Test values should be hard-coded when possible, to avoid weird/unexpected side effects.